lib/client/db/postgres/mcp/mcp.go

```diff
@@ -0,0 +1,245 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package mcp
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+	"github.com/mark3labs/mcp-go/mcp"
+
+	dbmcp "github.com/gravitational/teleport/lib/client/db/mcp"
+	"github.com/gravitational/teleport/lib/defaults"
+)
+
+// runQueryTool is the run query MCP tool definition.
+var runQueryTool = mcp.NewTool(dbmcp.ToolName(defaults.ProtocolPostgres, "run_query"),
```

Comment:
nit: `run_query` feels redundant, `query` by itself is probably fine, but up to you. Do we need `database` in the tool name like `teleport_database_postgres_query`.

And i assume we put database protocol before `run_query` is because other protocols may not use `query`?

 Naming is always the hardest part.

Reply:
context on this having `database` in prefix: we might need to ship Teleport Toolbox MCP with tools like `teleport_create_access_request`, `teleport_search_audit_events` in first release

Reply:
The point is to let the database protocol decide which "verb" to expose. For example, Redis might expose `set` and `get` operations, and MongoDB might expose `find` and `insert` operations instead of a single `query`.

Regarding the `database` prefix, it was primarily intended to avoid any potential conflicts. However, the database protocol may resolve this issue, because the possible future conflicts would also prefix their name:

```
teleport_postgres_query
teleport_redis_set
teleport_redis_get
teleport_ssh_exec
teleport_app_request
```

Reply:
I'm ok with having the `database_` prefix. What do you think @greedy52? Do you think the protocol is sufficient to differentiate these tools from others?

Reply:
looks good lets just protocol for now without the prefix